community/detectors/atlassian_confluence_cve_2023_22518/src/main/java/com/google/tsunami/plugins/detectors/rce/cve202322518/Cve202322518VulnDetector.java

```diff
@@ -0,0 +1,175 @@
+    return String.format("http://%s/", toUriAuthority(networkService.getNetworkEndpoint()));
+  }
+
+  private boolean isServiceVulnerable(NetworkService networkService) {
+    final String rootUri = buildRootUri(networkService);
+    final String targetUploadUri = rootUri + FILE_UPLOAD_PATH;
+
+    try {
+      MultipartBody mBody =
+          new MultipartBody.Builder()
+              .setType(MultipartBody.FORM)
+              .addFormDataPart("buildIndex", "false")
+              .addFormDataPart(
+                  "file",
+                  "someRandomStr14212134.zip",
+                  RequestBody.create(MediaType.parse("application/zip"), "someRandomStr14212134"))
+              .addFormDataPart("edit", "Upload and import")
+              .build();
+
+      Buffer sink = new Buffer();
+      mBody.writeTo(sink);
+
+      HttpResponse response =
+          httpClient.send(
+              post(targetUploadUri)
+                  .setHeaders(
+                      HttpHeaders.builder()
+                          .addHeader(
+                              CONTENT_TYPE, Objects.requireNonNull(mBody.contentType()).toString())
+                          .addHeader("X-Atlassian-Token", "no-check")
+                          .build())
+                  .setRequestBody(ByteString.copyFrom(sink.readByteArray()))
+                  .build(),
+              networkService);
+
+      String body;
+      if (response.bodyString().isPresent()) {
+        body = response.bodyString().get();
+        if (response.status().code() == HttpStatus.OK.code()
+            && body.contains("The zip file did not contain an entry")
+            && body.contains("exportDescriptor.properties")) {
+          return true;
+        }
+      } else {
+        return false;
+      }
+
+    } catch (Exception e) {
+      logger.atWarning().log("Failed to send request.");
+      logger.atWarning().log(e.toString());
+      logger.atWarning().log(e.getMessage());
+      return false;
+    }
+    return false;
+  }
+
+  private DetectionReport buildDetectionReport(
+      TargetInfo targetInfo, NetworkService vulnerableNetworkService) {
+    return DetectionReport.newBuilder()
+        .setTargetInfo(targetInfo)
+        .setNetworkService(vulnerableNetworkService)
+        .setDetectionTimestamp(Timestamps.fromMillis(Instant.now(utcClock).toEpochMilli()))
+        .setDetectionStatus(DetectionStatus.VULNERABILITY_VERIFIED)
+        .setVulnerability(
+            Vulnerability.newBuilder()
+                .setMainId(
+                    VulnerabilityId.newBuilder()
+                        .setPublisher("TSUNAMI_COMMUNITY")
+                        .setValue("CVE-2023-22518"))
+                .setSeverity(Severity.CRITICAL)
+                .setTitle("Atlassian Confluence Data Center Improper Authorization CVE-2023-22515")
+                .setDescription(
+                    "All versions of Confluence Data Center and Server are affected by this vulnerability."
```

Comment:
This is a bit misleading: this seems to indicate that there is no possible recommendation (i.e. "all versions are affected"). Could you please rework the description? Maybe skipping completely the first sentence.